### ISSUE TITLE
Map upload has unexpected behavior

### ISSUE THREAD

**Issue by Emptiest** [Quaver/Quaver#3045]
**Describe the bug**
I have a [~4 month old map](https://quavergame.com/mapset/map/36889) which I recently made changes to and attempted to upload. However, even after uploading the map still [wants to update](https://cdn.discordapp.com/attachments/629908083727925249/850503843002187827/unknown.png). The issue may be specific to this one mapset, however. There are some other weird things I noticed with this interaction, which I will detail below.

**To Reproduce**
1. Upload mapset

**Expected behavior**
1. Map on the website updates
2. Profile activity feed updates with "Name has updated mapset"
3. Mapset page updates the map
4. Local copy of the map should match what's on the server

**Screenshots**
The following screenshots were all taken immediately after uploading this map.

In-game
![](https://cdn.discordapp.com/attachments/629908083727925249/850503843002187827/unknown.png)

Mapset page
![](https://cdn.discordapp.com/attachments/629908083727925249/850504042517233704/unknown.png)

Activity feed
![](https://cdn.discordapp.com/attachments/629908083727925249/850504069154865162/unknown.png)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Edge
 - Version 0.29.0

**Additional context**
Before I noticed anything had gone wrong, I actually did upload this mapset about 21 hours ago, as the second screenshot indicates. However, there is no corresponding line in my activity feed. [Link to my profile](https://quavergame.com/user/2696) for convenience.

Not really sure what's causing this issue. Also not sure if the problem is the game or the website. Filing the issue here for now though.

**Comment by Emptiest** [comment]
Made no changes and just hit upload mapset again, and for some reason it's fine now. I don't really understand why it would resolve itself after a 3rd try, when the first two upload attempts went awry.

**Comment by Emptiest** [comment]
It appears the map that I just uploaded was the version of the map before I'd made changes 4 days ago. Throughout this process I'd been very careful not to update my local copy of the map from the one on the website, trusting that the version I had locally would still be the correct one, but after uploading it again today and checking the map afterwards, it appears the map somehow reverted to what I had before I'd made changes. That's about 3 hours of effort flushed...

I'm baffled and rather discouraged by this turn of events. I'm also afraid that by updating the mapset I've made this issue much more difficult to track down. Alas.
